### PR TITLE
chore(docs): Add mentioning of HOSTNAME env variable for standalone output

### PR DESCRIPTION
**Pull request by steppefox**
### What?

Update documentation to mention a `HOSTNAME` env variable for `standalone` output

Related to Feature: https://github.com/vercel/next.js/pull/44627 

### Why?

Current documentation mentions only `PORT` env, but does not mention `HOSTNAME` added in January

Affected documentation pages:

- https://nextjs.org/docs/app/api-reference/next-config-js/output

Affected examples:

- https://github.com/vercel/next.js/tree/canary/examples/with-docker
- https://github.com/vercel/next.js/tree/canary/examples/with-docker-multi-env

### How?

- Run examples
- Check documentation

### CODE REVIEW

File: examples/with-docker/Dockerfile [hunk 1]
```diff
@@ -55,5 +55,6 @@ USER nextjs
 EXPOSE 3000
 
 ENV PORT 3000
+ENV HOSTNAME 0.0.0.0
```
[the remark this answers is not in the document]
Comment (steppefox) [reply examples/with-docker/Dockerfile:58]: Can do that 👍 Had initial idea to show how to set alternative hostname. Do you mind if I change in the documentation then to `PORT=3000 HOSTNAME=0.0.0.0 node server.js`?

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 8326a60c295661358b554f8e9847a1ee81302a0a

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: examples/with-docker/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (steppefox) [reply examples/with-docker/Dockerfile]: Updated

File: docs/02-app/02-api-reference/05-next-config-js/output.mdx [hunk 2]
```diff
@@ -41,6 +41,7 @@ Additionally, a minimal `server.js` file is also output which can be used instea
 >
 > - `next.config.js` is read during `next build` and serialized into the `server.js` output file. If the legacy [`serverRuntimeConfig` or `publicRuntimeConfig` options](/docs/app/api-reference/next-config-js/runtime-configuration) are being used, the values will be specific to values at build time.
 > - If your project uses [Image Optimization](/docs/app/building-your-application/optimizing/images) with the default `loader`, you must install `sharp` as a dependency:
+> - If your project needs alternative port or hostname for listening, you can define `PORT` and `HOSTNAME` environment variables, before running `server.js`. For example, `PORT=3000 HOSTNAME=0.0.0.0 node server.js`.
```
Comment (styfle) [inline docs/02-app/02-api-reference/05-next-config-js/output.mdx:44]:
```suggestion
> - If your project needs alternative port or hostname for listening, you can define `PORT` and `HOSTNAME` environment variables, before running `server.js`. For example, `PORT=3000 HOSTNAME=localhost node server.js`.
```

Lets use the defaults here and the reader can decide to change if they want

Comment (steppefox) [reply docs/02-app/02-api-reference/05-next-config-js/output.mdx]: Ok 😢

**Review by styfle — APPROVED** [review]

**Pull request merged**